nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+            case STRING:
+            default:
+                return generateRandomString();
+        }
+    }
+
+    private String generateRandomString() {
+        final int categoryChoice = faker.number().numberBetween(0, 10);
+        switch (categoryChoice) {
+            case 0:
+                return faker.name().fullName();
+            case 1:
+                return faker.lorem().word();
+            case 2:
+                return faker.shakespeare().romeoAndJulietQuote();
+            case 3:
+                return faker.educator().university();
+            case 4:
+                return faker.zelda().game();
+            case 5:
+                return faker.company().name();
+            case 6:
+                return faker.chuckNorris().fact();
+            case 7:
+                return faker.book().title();
+            case 8:
+                return faker.dog().breed();
+            default:
+                return faker.animal().name();
+        }
+    }
+
+    protected RecordSchema generateRecordSchema(final Map<String, String> fields, final boolean nullable) {
+        final List<RecordField> recordFields = new ArrayList<>(fields.size());
+        for (Map.Entry<String, String> field : fields.entrySet()) {
+            final String fieldName = field.getKey();
+            final String fieldType = field.getValue();
+            final DataType fieldDataType = getDataType(fieldType);
+            RecordField recordField = new RecordField(fieldName, fieldDataType, nullable);
+            recordFields.add(recordField);
+        }
+        return new SimpleRecordSchema(recordFields);
+    }
+
+    // This method identifies "segments" by splitting the given name on underscores, then capitalizes each segment and removes the underscores. Ex: 'game_of_thrones' = 'GameOfThrones'
+    private static String normalizeClassName(String name) {
+        String[] segments = name.split("_");
+        String newName = Arrays.stream(segments).map((s) -> s.substring(0, 1).toUpperCase() + s.substring(1)).collect(Collectors.joining());
+        return newName;
+    }
+
+    // This method lowercases the first letter of the given name in order to match the name to a Faker method
+    private static String normalizeMethodName(String name) {
+
+        String newName = name.substring(0, 1).toLowerCase() + name.substring(1);
+        return newName;
+    }
+
+    // This method splits the given name on uppercase letters, ensures the first letter is capitalized, then joins the segments using a space. Ex. 'gameOfThrones' = 'Game Of Thrones'
+    private static String normalizeDisplayName(String name) {
+        // Split when the next letter is uppercase
+        String[] upperCaseSegments = name.split("(?=\\p{Upper})");
+
+        return Arrays.stream(upperCaseSegments).map(
+                (upperCaseSegment) -> upperCaseSegment.substring(0, 1).toUpperCase() + upperCaseSegment.substring(1)).collect(Collectors.joining(" "));
+    }
```

Comment:
Moving out these normalization methods into a separate testable interface and implementation would help streamline the Processor code and abstract the behavior.